src/sentry/api/endpoints/organization_dashboard_details.py

```diff
@@ -1,11 +1,98 @@
 from __future__ import absolute_import
 
+from django.db.models import Max
+from rest_framework import serializers
+from rest_framework.response import Response
+
 from sentry.api.base import DocSection
 from sentry.api.bases.dashboard import (
     OrganizationDashboardEndpoint
 )
 from sentry.api.serializers import serialize
-from sentry.models import ObjectStatus
+from sentry.api.serializers.rest_framework import ListField, ValidationError
+from sentry.models import ObjectStatus, Widget
+
+
+def get_next_dashboard_order(dashboard_id):
+    max_order = Widget.objects.filter(
+        dashboard_id=dashboard_id,
+    ).aggregate(Max('order'))['order__max']
+
+    return max_order + 1 if max_order else 1
+
+
+def remove_widgets(dashboard_widgets, widget_data):
+    """
+    Removes current widgets belonging to dashboard not in widget_data.
+    Returns remaining widgets.
+    """
+    widget_ids = [wd['id'] for wd in widget_data]
+    dashboard_widgets.exclude(
+        id__in=widget_ids
+    ).update(status=ObjectStatus.PENDING_DELETION)
```

Comment:
What actually deletes these records?

Reply:
Oops you're right. In this case I should just delete.